src/Castle.Windsor/Core/Internal/ReflectionUtil.cs

```diff
@@ -307,7 +308,7 @@ private static AssemblyName GetAssemblyName(string filePath)
 		private static TBase Instantiate<TBase>(Type subtypeofTBase, object[] ctorArgs)
 		{
 			ctorArgs = ctorArgs ?? new object[0];
-			var types = ctorArgs.ConvertAll(a => a == null ? typeof(object) : a.GetType());
```

Comment:
Have you considered creating a shim in windsor for it?

Something like this: https://github.com/cryosharp/fortress/blob/master/src/Fortress.Windsor/Compatibility/EnumerableExtensions.cs#L22